iModelCore/ECDb/ECDb/SchemaWriter.cpp

```diff
@@ -3262,22 +3262,31 @@ BentleyStatus SchemaWriter::DeleteCustomAttributeClass(Context& ctx, ECCustomAtt
 //---------------------------------------------------------------------------------------
 // @bsimethod
 //+---------------+---------------+---------------+---------------+---------------+------
-BentleyStatus SchemaWriter::DeleteClass(Context& ctx, ClassChange& classChange, ECClassCR deletedClass)
+BentleyStatus SchemaWriter::DeleteClass(Context& ctx, ClassChange& classChange, ECClassCR deletedClass, bool isDynamicSchema)
     {
+    auto isMajorSchemaVersionChange = ctx.IsMajorSchemaVersionChange(deletedClass.GetSchema().GetId());
+
     if (!ctx.AreMajorSchemaVersionChangesAllowed() || !ctx.IsMajorSchemaVersionChange(deletedClass.GetSchema().GetId()))

```

Comment:
you store isMajorSchemaVersionChange but then call it again here.  If you're going to cache it use it everywhere